crates/iroha_genesis/src/lib.rs

```diff
@@ -356,6 +389,53 @@ impl Decode for ExecutorPath {
     }
 }
 
+/// For readability of [`RawGenesisTransaction`], briefly denotes a trigger with wasm executable registered in genesis
+#[derive(Debug, Clone, Serialize, Deserialize, IntoSchema, Encode, Decode, Constructor)]
+pub struct GenesisWasmTrigger {
+    id: TriggerId,
+    action: GenesisWasmAction,
+}
+
+/// For readability of [`GenesisWasmTrigger`], briefly denotes an action with wasm executable
+#[derive(Debug, Clone, Serialize, Deserialize, IntoSchema, Encode, Decode)]
+pub struct GenesisWasmAction {
+    /// Expected to be converted by [`iroha_test_samples::load_sample_wasm`]
+    executable: String,
+    repeats: Repeats,
+    authority: AccountId,
+    filter: EventFilterBox,
+}
+
+impl GenesisWasmAction {
+    /// Construct [`GenesisWasmAction`]
+    pub fn new(
+        executable: impl ToString,
+        repeats: impl Into<Repeats>,
+        authority: AccountId,
+        filter: impl Into<EventFilterBox>,
+    ) -> Self {
+        Self {
+            executable: executable.to_string(),
+            repeats: repeats.into(),
+            authority,
+            filter: filter.into(),
+        }
+    }
+}
+
+impl From<GenesisWasmTrigger> for Trigger {
+    fn from(src: GenesisWasmTrigger) -> Self {
+        Trigger::new(src.id, src.action.into())
+    }
+}
+
+impl From<GenesisWasmAction> for Action {
+    fn from(src: GenesisWasmAction) -> Self {
+        let executable = load_sample_wasm(src.executable);
```

Comment:
All paths in configuration are relative to the source file they are specified at.

Please make sure that this path is resolved relatively to the path of `genesis.json` itself.